### ISSUE TITLE
UNUSED_LOCAL_VARIABLE false positive in loop

### ISSUE THREAD

**Issue by jaredbeck** [cflint/CFLint#631]
```
<cffunction name="getDataForExport" returntype="array" output="false" access="public">
  <!--- ... --->
  <cfscript>
    // ...
    var thisBase = getRowBaseFromMap(baseDataMap, row);
    for( var v in baseVars ){
      ar.Append(thisBase.KeyExists(v.shortname) ? thisBase[v.shortname] : '');
    }
    // ...
```

```
Message code:UNUSED_LOCAL_VARIABLE
        File:/Users/jared/redacted/redacted.cfc
        Column:8
        Line:135
                Message:Local variable thisBase is not used in function getDataForExport. Consider removing it.
                Variable:'thisBase' in function: getDataForExport
```

This issue is distinct from https://github.com/cflint/CFLint/issues/211, but it may be helpful to review that fix.

**Comment by KamasamaK** [comment]
Even though it may be a distinct issue, #211 doesn't appear to be working in this example either. I am also getting `Local variable v is not used in function getDataForExport.`

**Comment by ryaneberly** [comment]
https://github.com/cfparser/cfparser/issues/125

**Comment by ryaneberly** [comment]
fixed with migration to cfparser 2.10.0.  tests added to dev branch
